File: backend/app/api/v1/quotes.py
import datetime

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependencies import get_db
from app.schemas.quote import QuoteStats, QuoteTimeSeries
from app.services.quote_service import QuoteService

router = APIRouter(prefix="/securities", tags=["Quotes"])


@router.get("/{security_id}/quotes", response_model=QuoteTimeSeries)
async def get_quotes(
    security_id: int,
    from_date: datetime.date | None = Query(None, description="Start date"),
    to_date: datetime.date | None = Query(None, description="End date"),
    interval: str = Query("day", description="Aggregation interval: day, week, month"),
    db: AsyncSession = Depends(get_db),
):
    service = QuoteService(db)
    return await service.get_quotes(
        security_id=security_id,
        from_date=from_date,
        to_date=to_date,
        interval=interval,
    )


@router.get("/{security_id}/quotes/stats", response_model=QuoteStats)
async def get_quote_stats(
    security_id: int,
    db: AsyncSession = Depends(get_db),
):
    service = QuoteService(db)
    return await service.get_stats(security_id)

File: frontend/src/api/quotes.ts
import apiClient from './client';
import type { QuoteTimeSeries, QuoteStats } from '@/types/quote';

export interface QuoteParams {
  from_date?: string;
  to_date?: string;
  interval?: string;
}

export const quotesApi = {
  getQuotes(
    securityId: number,
    params?: QuoteParams,
  ): Promise<QuoteTimeSeries> {
    return apiClient.get(`/quotes/security/${securityId}`, {
      params,
    }) as Promise<QuoteTimeSeries>;
  },

  getStats(securityId: number): Promise<QuoteStats> {
    return apiClient.get(
      `/quotes/security/${securityId}/stats`,
    ) as Promise<QuoteStats>;
  },
};
